File: facade/Facade.py


class TXTFile:
    pass
class CompressCodec:
    pass

class CodecFactory:
    def convert(self, cc:CompressCodec):
        return PDFFile();

class PDFFile :
    pass
class FileConvertFactory :
    def convert(self, txtFile: TXTFile) ->PDFFile:
        compressCodec =  CompressCodec();
        cf =  CodecFactory();
        # do something
        return cf.convert(compressCodec);

# Client
if __name__ == "__main__":
    ff = FileConvertFactory()
    ff.convert(TXTFile())

File: facade/Facade.ts
class TXTFile {
}

class CompressCodec {
}

class PDFFile {
}

class CodecFactory {
    convert(cc: CompressCodec): PDFFile {
        return new PDFFile();
    }
}


class FileConvertFactory {
    public convert(txtFile: TXTFile): PDFFile {
        const compressCodec: CompressCodec = new CompressCodec();
        const cf: CodecFactory = new CodecFactory();
        // do something
        return cf.convert(compressCodec);
    }
}

const ff = new FileConvertFactory();
ff.convert(new TXTFile())